src/pool-cl/CLPositionManager.sol

```diff
@@ -235,6 +254,31 @@ contract CLPositionManager is
         (liquidityDelta - feesAccrued).validateMaxIn(amount0Max, amount1Max);
     }
 
+    /// @dev The liquidity delta is derived from open deltas in the pool manager.
+    function _increaseFromDeltas(uint256 tokenId, uint128 amount0Max, uint128 amount1Max, bytes calldata hookData)
+        internal
+        onlyIfApproved(msgSender(), tokenId)
+    {
+        (PoolKey memory poolKey, CLPositionInfo info) = getPoolAndPositionInfo(tokenId);
+
+        (uint160 sqrtPriceX96,,,) = clPoolManager.getSlot0(poolKey.toId());
+
+        // Use the credit on the pool manager as the amounts for the mint.
+        uint256 liquidity = LiquidityAmounts.getLiquidityForAmounts(
+            sqrtPriceX96,
+            TickMath.getSqrtRatioAtTick(info.tickLower()),
+            TickMath.getSqrtRatioAtTick(info.tickUpper()),
+            _getFullCredit(poolKey.currency0),
+            _getFullCredit(poolKey.currency1)
+        );
```

Comment:
Add a new function , so can reuse this part , will reduce contract size and increase a little gas.
https://github.com/pancakeswap/pancake-v4-periphery/pull/64

Reply:
consider merging separately per discussion